SmartTutor/ContentModel/ChallengeService.cs

```diff
@@ -0,0 +1,105 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using RepositoryCompiler.Controllers;
+using SmartTutor.ContentModel.LearningObjects;
+using SmartTutor.ContentModel.LearningObjects.Repository;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SmartTutor.ContentModel
+{
+    public class ChallengeService : IChallengeService
+    {
+        private readonly ILearningObjectRepository _learningObjectRepository;
+
+        public ChallengeService(ILearningObjectRepository learningObjectRepository)
+        {
+            _learningObjectRepository = learningObjectRepository;
+        }
+
+        public ChallengeService() { }
+
+        internal Dictionary<string, double> GetMetricNamesFromClasses(List<CaDETClass> caDETClasses)
```

Comment:
This type of logic is decidedly not coordination logic so we should think about what it does and where it fits better.

Notably, we will change the RepoCompiler CaDETModel so that the metrics and their values are key value pairs. Essentially, each CaDETClass will have a "Dictionary<string, double> Metrics".

Getting the names (althought I'm not sure why we need the names?) would then be trivial (e.g., Metrics.Keys)